src/main/java/com/epam/aidial/core/controller/RateResponseController.java

```diff
@@ -0,0 +1,178 @@
+package com.epam.aidial.core.controller;
+
+import com.epam.aidial.core.Proxy;
+import com.epam.aidial.core.ProxyContext;
+import com.epam.aidial.core.config.Assistant;
+import com.epam.aidial.core.config.Config;
+import com.epam.aidial.core.config.Deployment;
+import com.epam.aidial.core.util.BufferingReadStream;
+import com.epam.aidial.core.util.HttpStatus;
+import com.epam.aidial.core.util.ProxyUtil;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpClientRequest;
+import io.vertx.core.http.HttpClientResponse;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.core.http.RequestOptions;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+
+import java.net.URL;
+
+import static com.epam.aidial.core.controller.DeploymentPostController.ASSISTANT;
+
+@Slf4j
+@RequiredArgsConstructor
+public class RateResponseController {
+
+    private final Proxy proxy;
+    private final ProxyContext context;
+
+    public void handle(String deploymentId) {
+        Deployment deployment = getDeployment(deploymentId);
+
+        if (deployment == null || !DeploymentController.hasAccessByUserRoles(context, deployment)) {
+            context.respond(HttpStatus.FORBIDDEN, "Forbidden deployment");
+            return;
+        }
+
+        if (deployment.getRateEndpoint() == null) {
+            context.getResponse().setStatusCode(200);
+            context.getResponse().send();
+            proxy.getLogStore().save(context);
```

Comment:
Why don't you use shortcut context.respond(HttpStatus.OK)?

Reply:
missed the method. fixed.